File: packages/SyncCore/Sources/SyncCore/DestinationStorageMode.swift
/// How received resources are arranged inside the chosen destination.
///
/// All layouts keep the fixed `iPhoneSync` container that lives directly under
/// the user-selected destination root. The cases below describe what appears
/// inside that container.
public enum DestinationStorageMode: String, Codable, Equatable, Sendable {
    /// `<destination>/iPhoneSync/<album>/<year>/<month>/<file>`
    case albumDate
    /// `<destination>/iPhoneSync/<album>/<file>`
    case albumOnly
    /// `<destination>/iPhoneSync/<file>`
    case flat
}

File: packages/SyncCore.Windows/src/receiver/destination-storage-mode.ts
// DestinationStorageMode — enum mirroring `DestinationStorageMode.swift`.

export const DestinationStorageMode = {
  albumDate: 'albumDate',
  albumOnly: 'albumOnly',
  flat: 'flat',
} as const;

export type DestinationStorageModeType = typeof DestinationStorageMode[keyof typeof DestinationStorageMode];

export function isDestinationStorageMode(value: string): value is DestinationStorageModeType {
  return value === DestinationStorageMode.albumDate
    || value === DestinationStorageMode.albumOnly
    || value === DestinationStorageMode.flat;
}
